hack/e2e.sh

```diff
@@ -29,6 +29,21 @@ set -o pipefail
 REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
 cd "${REPO_ROOT}" || exit 1
 
+# If kind is not available on the path, get it
+if ! [ -x "$(command -v kind)" ]; then
+  echo 'kind not found, installing'
+  GO111MODULE="on" go get sigs.k8s.io/kind@v0.4.0
+fi
+
+# If kubectl is not available on the path, get it
+# TODO: adapt for other OS and ARCH
+if ! [ -x "$(command -v kubectl)" ]; then
+  echo 'kubectl not found, installing'
+  curl -LO https://storage.googleapis.com/kubernetes-release/release/v1.15.0/bin/linux/amd64/kubectl
+  chmod +x kubectl
+  mv kubectl ~/bin
```

Comment:
Need to have a better location for downloading to, since this will likely not work in prow

Reply:
@chuckha @ncdc what are your thoughts on making some of the testing prerequisites like `kubectl`, `kustomize`, and `kind` development prerequisites and us building a custom image for use in prow with them already baked in? Then we could have a script that just ensures they are >= a particular version and avoid having to download/build within bazel or do this type of downloading shenanigans in the test wrappers.

Reply:
We could even potentially eliminate the `kustomize` requirement all together with a new enough version of `kubectl`

Reply:
Seems reasonable to me. And it would ensure a consistent experience (locked versions of the tools) across environments.